### DISCUSSION TITLE
Set up excalidraw and excalidraw-room on an EC2 instance and proxy the app using Cloudflare

### DISCUSSION THREAD

**Discussion by DineshCodeFlow** [excalidraw/excalidraw#8307] [General]
To set up excalidraw-room on an EC2 instance and proxy the app using Cloudflare, you need to:

1- Cloudflare doesn’t support all ports. I was running the Excalidraw collaboration app on a port not supported by Cloudflare, as we’re using CF as a proxy. I have now changed this to a port supported https://developers.cloudflare.com/fundamentals/reference/network-ports/.
2: Most importantly, Cloudflare certificates only support the top-level domain. I am running another Node app on the same EC2 instance but on a different port: XXXX. The URL is https://api.shorterloop.com:XXXX/. For this, I needed to generate different certificates.
3: We needed to set up our own Firebase account, we did that!

Posting this to help others in the community!

[Our whiteboard](https://app.shorterloop.com):
<img width="1245" alt="ShorterLoop Whiteboard" src="https://github.com/user-attachments/assets/9b31a1f4-06d4-4412-b2f1-09b3a19f5903">

**Comment by kkohli-prodeasy** [comment]
Unable to use real time updates
